Feature internationalization

koheitakahashi opened the pull request:
Internationalized app.

Commit:
[Add] Added dictionary files.

Commit:
[Change] Changed default locale and time zone.

Commit:
[Add] Added language switching link.

Commit:
[Change] Internationalized messages (flash, labels, confirm and buttons).

Commit:
[Add] Added RSpec for checking i18n.

Commit:
[Change] Erb file in views changed slim.

Commit:
[Change] Changed design by bootstrap.

Commit:
[Clean] Removed unnecessary files and lines.

Commit:
[Fix] Fixed with rubocop.

komagata commented on app/assets/stylesheets/application.scss:
```diff
@@ -0,0 +1 @@
+@import "bootstrap";
```
最終行に改行がないようです〜

komagata commented on app/controllers/books_controller.rb:
```diff
@@ -9,28 +9,19 @@ def index
       if @book.save
-        format.html { redirect_to @book, notice: "Book was successfully created." }
+        format.html { redirect_to @book, notice: "#{t 'errors.messages.Book_was_successfully_created.'}"  }
         format.json { render :show, status: :created, location: @book }
```
JSONを使う必要がなければ関連処理は削除しちゃってください〜

komagata commented on app/views/books/_form.html.slim:
```diff
@@ -0,0 +1,28 @@
+      ul
+        - book.errors.full_messages.each do |message|
+          li = message
+
```
二行空いてるのが気になりますね。

komagata commented on app/views/books/edit.html.slim:
```diff
@@ -0,0 +1,7 @@
+
+= link_to "#{t'link.Show'}", @book
+'|
+= link_to "#{t'link.Back'}", books_path
```
最終行に改行が無いようです〜

Commit:
[Fix] Fixed the parts indicated.

komagata commented on app/controllers/books_controller.rb:
```diff
@@ -9,32 +9,21 @@ def index
       if @book.save
-        format.html { redirect_to @book, notice: "Book was successfully created." }
-        format.json { render :show, status: :created, location: @book }
+        format.html { redirect_to @book, notice: "#{t 'errors.messages.Book_was_successfully_created.'}" }
```
他のフォーマットに対応しないのであればrespond_to自体が必要ないです。

respond_toがどういう機能を持つのか調べてみると良いかもです。

koheitakahashi replied:
承知しました‼️

調べてみて、`respond_to`は`format.{形式}`で指定された形式で情報を返す機能だと分かりました‼️

komagata commented on app/views/books/new.html.slim:
```diff
@@ -0,0 +1,6 @@
+= render 'form', book: @book
+
+.nav.justify-content-end
+  = link_to "#{t'link.Back'}", books_path
```
最終行に改行が無いようです。

komagata commented on config/locales/carriewave/en.yml:
```diff
@@ -0,0 +1,14 @@
+      rmagick_processing_error: "Failed to manipulate with rmagick, maybe it is not an image?"
+      mini_magick_processing_error: "Failed to manipulate with MiniMagick, maybe it is not an image? Original Error: %{e}"
+      min_size_error: "File size should be greater than %{min_size}"
+      max_size_error: "File size should be less than %{max_size}"
```
最終行に改行が無いようです。

komagata commented:
エディターで自動的に最終行に改行が入るように設定したほうがいいとおもいます。

また、ローカルで確認したとしても、PRでレビュー依頼する時は必ずFiles changedを確認してから出す癖をつけた方がいいとおもいます。仕事でやっている人はみなそういう癖が自然についている気がします〜

Commit:
Fixed the parts indicated.

In particular, Added a line at the end of the file and, Removed unnecessary description that "respond_to" related.

NMP300 commented:
> エディターで自動的に最終行に改行が入るように設定したほうがいいとおもいます。

教えていただき、ありがとうございます。エディタの方設定しました↓
<img width="383" alt="スクリーンショット 2019-11-11 14 07 26" src="https://user-images.githubusercontent.com/51867807/68562352-a3e75200-048c-11ea-912d-4adcfe8b797a.png">

> また、ローカルで確認したとしても、PRでレビュー依頼する時は必ずFiles changedを確認してから出す癖をつけた方がいいとおもいます。仕事でやっている人はみなそういう癖が自然についている気がします〜

チェックの手間を増やしてしまい申し訳ありません💦🙇‍♂️
レビューしていただく方の負担を増やさないようにFile changedは必ず確認したいと思います🔥

そこで、1点質問なのですが、File changedを確認するとき、@komagataさんはどのような点に気をつけてらっしゃいますか？自分なりのチェックリストのようなものがあるのでしょうか？

私の場合、なんとなく確認しているだけの感じがしてしまって、気をつけて見ることができていない気がしたので、教えていただければ幸いです🙇‍♂️

komagata commented:
最終行の改行やインデント、不要なコメントなどはパッと見てわかるのでそれはチェックしています。

自分がPRを作る場合はそれ以前にコードフォーマットや最終行の改行はツールでやっているで気にしていないです。

自分のPRを作ったらFiles Changedでみて、全ての行をどういう意図で書いてるのかを自分の中で考えながらチェックしています。

ただViewはあんまりちゃんとチェックしてないです。

komagata reviewed (approved):
確認しました、OKですー🙆‍♂️

NMP300 commented:
なるほど🤔
教えていただきありがとうございます🙇‍♂️

自分の中で、ここをチェックした方が良いというチェックリストができていなかったので、とても参考になります‼️
教えていただいた項目を中心に注意深く自分のコードをチェックしていきたいと思います🔥
